src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -0,0 +1,1433 @@
+          pullConfig = Option(DockerPullConfig("pullConfigSecret")))))
+      val app = App(id = "/app", cmd = Some("cmd"), container = Option(container))
+      val (body, plan) = prepareApp(app, groupManager)
+
+      When("The create request is made")
+      clock += 5.seconds
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("It fails")
+        status shouldEqual StatusCodes.UnprocessableEntity withClue s"body: ${ByteString(body).utf8String}, response: ${responseAs[String]}"
+
+        val responseStr = responseAs[String]
+        responseStr should include("/container/docker/pullConfig")
+        responseStr should include("must be empty")
+        responseStr should include("Feature secrets is not enabled. Enable with --enable_features secrets)")
+      }
+    }
+
+    "Fail creating application when network name is missing" in new Fixture {
+      Given("An app and group")
+      val app = App(
+        id = "/app",
+        cmd = Some("cmd"),
+        networks = Seq(Network(mode = NetworkMode.Container))
+      )
+
+      When("The create request is made")
+      val entity = HttpEntity(Json.stringify(Json.toJson(app)).getBytes("UTF-8")).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("Validation fails")
+        status shouldEqual StatusCodes.UnprocessableEntity
+        responseAs[String] should include(NetworkValidationMessages.NetworkNameMustBeSpecified)
+      }
+    }
+
+    "Create a new app with IP/CT, no default network name, Alice does not specify a network" in new Fixture {
+      Given("An app and group")
+      val app = App(
+        id = "/app",
+        cmd = Some("cmd"),
+        networks = Seq(Network(mode = NetworkMode.Container))
+      )
+      the[NormalizationException] thrownBy {
+        prepareApp(app, groupManager)
+      } should have message NetworkNormalizationMessages.ContainerNetworkNameUnresolved
+    }
+
+    "Create a new app with IP/CT on virtual network foo" in new Fixture {
+      Given("An app and group")
+      val app = App(
+        id = "/app",
+        cmd = Some("cmd"),
+        networks = Seq(Network(mode = NetworkMode.Container, name = Some("foo")))
+      )
+      val (body, plan) = prepareApp(app, groupManager)
+
+      When("The create request is made")
+      clock += 5.seconds
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("It is successful")
+        status shouldEqual StatusCodes.Created withClue s"body = ${ByteString(body).utf8String}, response = ${responseAs[String]}"
+        header[Headers.`Marathon-Deployment-Id`] should not be 'empty
+
+        And("the JSON is as expected, including a newly generated version")
+        import mesosphere.marathon.api.v2.json.Formats._
+        val expected = AppInfo(
```

Comment:
How about avoiding the serialization.

Reply:
I'm not sure it's possible because deployment id is generated dynamically.

Reply:
The can be easily set with string interpolation.